### PR TITLE
Assignment 5

### PR DESCRIPTION
**Pull request by JerettLatimer**

### CODE REVIEW

**Comment by mcstudor** [comment]
I will review this

File: SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace SecretSanta.Business.Dto
+{
+    public class UserInput
+    {
+        [Required]
+        public string? FirstName { get; set; }
+        [Required]
+        public string? LastName { get; set; }
+        public IList<Gift> Gifts { get; } = new List<Gift>();
+        public IList<UserGroup> UserGroups { get; } = new List<UserGroup>();
```
Comment (mcstudor) [inline SecretSanta/src/SecretSanta.Business/Dto/UserInput.cs]: These two line expose the entity objects to the controller. He discussed how our DTO objects should only contain primitives.

File: SecretSanta/src/SecretSanta.Api/Controllers/BaseApiController.cs [hunk 2]
```diff
@@ -35,13 +38,18 @@ public async Task<IActionResult> Get(int id)
         }
 
         [HttpPut("{id}")]
-        public async Task<TEntity?> Put(int id, [FromBody] TEntity value)
+        public async Task<IActionResult> Put(int id, [FromBody] TInputDto value)
         {
-            return await Service.UpdateAsync(id, value);
+            TDto tDto = await Service.UpdateAsync(id, value);
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Api/Controllers/BaseApiController.cs]: In general when naming you don't include the "t" from the generic parameter. 
```suggestion
            TDto dto = await Service.UpdateAsync(id, value);
```

File: SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            Assert.AreEqual(gift.Title, returnedGift.Title);
+            Assert.AreEqual(gift.Description, returnedGift.Description);
+            Assert.AreEqual(gift.Url, returnedGift.Url);
+            Assert.AreEqual(gift.UserId, returnedGift.UserId);
+        }
+
+        [TestMethod]
+        [DataRow(nameof(Business.Dto.GiftInput.Title))]
+        [DataRow(nameof(Business.Dto.GiftInput.UserId))]
+        public async Task Post_WithoutRequiredProperties_BadRequest(string propertyName)
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Api.Tests/Controllers/GiftControllerTests.cs]: Would have been good to do a Post test that succeeds.

**Pull request closed without merging**